graylog2-server/src/main/java/org/graylog2/indexer/fieldtypes/MappedFieldTypesServiceImpl.java

```diff
@@ -76,7 +80,11 @@ public Set<MappedFieldTypeDTO> fieldTypesByStreamIds(Collection<String> streamId
 
     private MappedFieldTypeDTO mapPhysicalFieldType(FieldTypeDTO fieldType) {
         final FieldTypes.Type mappedFieldType = fieldTypeMapper.mapType(fieldType).orElse(UNKNOWN_TYPE);
-        return MappedFieldTypeDTO.create(fieldType.fieldName(), mappedFieldType);
+        return new MappedFieldTypeDTO(fieldType.fieldName(),
+                mappedFieldType,
+                fieldUnitObtainer.obtainUnit(fieldType.fieldName())
+                        .map(unit -> new UnitId(unit.unitType(), unit.abbrev()))
```

Comment:
As this part is duplicated, I think we should introduce a factory method in `UnitId` that allows constructing it from a `Unit` instance and/or an overloaded factory method in `MappedFieldTypeDTO` that takes a `Unit` as a third parameter.